packages/gen-ai-hub/src/client/openai/openai-client.ts

```diff
@@ -1,60 +1,107 @@
+import { HttpRequestConfig } from '@sap-cloud-sdk/http-client';
+import { CustomRequestConfig, executeRequest } from '@sap-ai-sdk/core';
 import {
-  BaseLlmParameters,
-  CustomRequestConfig,
-  executeRequest
-} from '@sap-ai-sdk/core';
-import { BaseClient } from '../interface.js';
+  DeploymentResolver,
+  resolveDeployment
+} from '../../utils/deployment-resolver.js';
 import {
   OpenAiChatCompletionParameters,
   OpenAiEmbeddingParameters,
   OpenAiEmbeddingOutput,
-  OpenAiChatCompletionOutput
+  OpenAiChatCompletionOutput,
+  OpenAiChatModel,
+  OpenAiEmbeddingModel
 } from './openai-types.js';
 
 const apiVersion = '2024-02-01';
 
 /**
- * OpenAI GPT Client.
+ * OpenAI Client.
  */
-export class OpenAiClient implements BaseClient<BaseLlmParameters> {
+export class OpenAiClient {
   /**
    * Creates a completion for the chat messages.
+   * @param model - The model to use for the chat completion.
    * @param data - The input parameters for the chat completion.
+   * @param deploymentResolver - A deployment id or a function to retrieve it.
    * @param requestConfig - The request configuration.
    * @returns The completion result.
    */
   async chatCompletion(
+    model: OpenAiChatModel | { name: OpenAiChatModel; version: string },
     data: OpenAiChatCompletionParameters,
+    deploymentResolver?: DeploymentResolver,
     requestConfig?: CustomRequestConfig
   ): Promise<OpenAiChatCompletionOutput> {
+    const deploymentId = await resolveOpenAiDeployment(
+      model,
+      deploymentResolver
+    );
     const response = await executeRequest(
       {
-        url: `/inference/deployments/${data.deploymentConfiguration.deploymentId}/chat/completions`,
+        url: `/inference/deployments/${deploymentId}/chat/completions`,
         apiVersion
       },
       data,
-      requestConfig
+      mergeRequestConfig(requestConfig)
```

Comment:
[q] Aren't we calling merge twice? Once here and then again in `executeRequest`?